### PR TITLE
Modify QueryAsyncVolume API unit test to be invoked only for vSphere 7.0.3 and above

### PR DESCRIPTION
**Pull request by chethanv28**
Recently we added go bindings and updated unit tests to invoke QueryAsync API in `cns/client_test.go`. However, we want to test invocations of the new API only for vSphere version 7.0.3 and above. Because, the API is supported only for vSphere versions 7.0.3 or later. This PR is addressing this issue by adding a vSphere version check before calling the API

Testing done:
Ran the unit tests with latest vSphere version
<pre>
go test -v
=== RUN   TestClient
    client_test.go:990: Successfully queried Volume using queryAsync API. queryVolumeAsyncTaskResult: &types.CnsAsyncQueryResult{
            CnsVolumeOperationResult: types.CnsVolumeOperationResult{},
            QueryResult:              types.CnsQueryResult{
                Volumes: []types.CnsVolume{
                    {
                        VolumeId: types.CnsVolumeId{
                            Id: "58e7c9e6-7c89-42d7-8c50-820e6011dd30",
                        },
                        DatastoreUrl:         "",
                        Name:                 "pvc-1296086c-282e-410d-b15a-7ebc6b0e8dcc",
                        VolumeType:           "BLOCK",
                        StoragePolicyId:      "",
                        Metadata:             types.CnsVolumeMetadata{},
                        BackingObjectDetails: &types.CnsBlockBackingDetails{
                            CnsBackingObjectDetails: types.CnsBackingObjectDetails{
                                CapacityInMb: 10,
                            },
                            BackingDiskId:      "58e7c9e6-7c89-42d7-8c50-820e6011dd30",
                            BackingDiskUrlPath: "",
                        },
                        ComplianceStatus:             "nonCompliant",
                        DatastoreAccessibilityStatus: "notApplicable",
                        HealthStatus:                 "",
                    },
                },
                Cursor: types.CnsCursor{
                    Offset:       1,
                    Limit:        100,
                    TotalRecords: 1,
                },
            },
        }
--- PASS: TestClient (2.78s)
PASS
ok      github.com/vmware/govmomi/cns   2.966s
</pre>

Ran the same unit tests on older vSphere versions, the test was skipped
<pre>
 go test -v
=== RUN   TestClient
--- PASS: TestClient (0.24s)
PASS
ok      github.com/vmware/govmomi/cns   0.428s
</pre>

### CODE REVIEW

**Comment by embano1** [comment]
@chethanv28 is there an issue related to this PR? Can you please provide a short description of the fix/feature? See our CONTRIB guidelines for details.

**Comment by chethanv28** [comment]
> @chethanv28 is there an issue related to this PR? Can you please provide a short description of the fix/feature? See our CONTRIB guidelines for details.

My bad, I forgot to update the MR description. Updated the same.

**Comment by chethanv28** [comment]
cc: @dougm

File: cns/client_test.go [hunk 1]
```diff
@@ -962,27 +967,57 @@ func TestClient(t *testing.T) {
...
-		if queryVolumeAsyncOperationRes.Fault != nil {
-			t.Fatalf("Failed to query volumes with QueryVolumeAsync. fault=%+v", queryVolumeAsyncOperationRes.Fault)
+}
+
+// isvSphereVersion70U3orAbove checks if specified version is 7.0 Update 3 or higher
+// The method takes aboutInfo{} as input which contains details about
+// VC version, build number and so on.
+// If the version is 7.0 Update 3 or higher, the method returns true, else returns false
+// along with appropriate errors during failure cases
+func isvSphereVersion70U3orAbove(ctx context.Context, aboutInfo types.AboutInfo) bool {
```
Comment (dougm) [inline cns/client_test.go:1002]: I think we ought to move the [govc version helpers](https://github.com/vmware/govmomi/blob//govc/flags/version.go#L51) to another package so you could make use of them here.  There's other use cases for it too.  But this is fine for now.

**Review by dougm — APPROVED** [review]
lgtm @chethanv28 
Also added you to the GH team, so you're able to merge changes yourself with one or more team member approval.

**Comment by chethanv28** [comment]
> lgtm @chethanv28
> Also added you to the GH team, so you're able to merge changes yourself with one or more team member approval.

Thanks @dougm

File: cns/client_test.go (hunk 1 above)
Comment (chethanv28) [reply cns/client_test.go:1002]: Sounds good. Will follow-up with govc version helpers in another PR

**Review by divyenpatel — APPROVED** [review]
/approve

**Review by lintongj — APPROVED** [review]
/lgtm

**Comment by chethanv28** [comment]
@dougm Can you approve this again. I added another fix in this PR and lost your approval

**Review by dougm — APPROVED** [review]

**Pull request merged**